src/middleware.js

```diff
@@ -1,38 +1,42 @@
 import * as defaultOptions from './defaults';
 import { getActionTypes } from './getActionTypes';
 
-function addInterceptor(target, candidate, getState) {
+function addInterceptor(target, candidate, injectedParameters) {
   if (!candidate) return;
   const successInterceptor = typeof candidate === 'function' ? candidate : candidate.success;
   const errorInterceptor = candidate && candidate.error;
-  target.use(successInterceptor && successInterceptor.bind(null, getState),
-    errorInterceptor && errorInterceptor.bind(null, getState));
+  target.use(successInterceptor && successInterceptor.bind(null, injectedParameters),
+    errorInterceptor && errorInterceptor.bind(null, injectedParameters));
 }
 
-function bindInterceptors(client, getState, middlewareInterceptors = {}, clientInterceptors = {}) {
+function bindInterceptors(client, injectedParameters, middlewareInterceptors = {}, clientInterceptors = {}) {
   [...middlewareInterceptors.request || [], ...clientInterceptors.request || []].forEach((interceptor) => {
-    addInterceptor(client.interceptors.request, interceptor, getState);
+    addInterceptor(client.interceptors.request, interceptor, injectedParameters);
   });
   [...middlewareInterceptors.response || [], ...clientInterceptors.response || []].forEach((interceptor) => {
-    addInterceptor(client.interceptors.response, interceptor, getState);
+    addInterceptor(client.interceptors.response, interceptor, injectedParameters);
   });
 }
 
 export const multiClientMiddleware = (clients, customMiddlewareOptions) => {
   const middlewareOptions = { ...defaultOptions, ...customMiddlewareOptions };
   const setupedClients = {};
+  let storedAction;
   return ({ getState, dispatch }) => next => action => {
     if (!middlewareOptions.isAxiosRequest(action)) {
       return next(action);
     }
+    storedAction = action;
     const clientName = middlewareOptions.getClientName(action) || middlewareOptions.defaultClientName;
     if (!clients[clientName]) {
       throw new Error(`Client with name "${clientName}" has not been defined in middleware`);
     }
     if (!setupedClients[clientName]) {
       const clientOptions = { ...middlewareOptions, ...clients[clientName].options };
       if (clientOptions.interceptors) {
-        bindInterceptors(clients[clientName].client, { getState, dispatch, action },
+        const getAction = () => storedAction;
```

Comment:
Perhaps I'm missing something, but I'm not sure that this is doing what it should.  It looks to me like `getAction` will return the most recent "axios action" dispatched through any of the clients; not necessarily the action which initiated the request.

As another action could be dispatched whilst waiting for the response, we don't necessarily have the correct action in the response interceptor.

Am I missing something obvious here?